lib/libzpool/kernel.c

```diff
@@ -813,11 +813,9 @@ random_get_pseudo_bytes(uint8_t *ptr, size_t len)
 int
 ddi_strtoull(const char *str, char **nptr, int base, u_longlong_t *result)
 {
-	(void) nptr;
-	char *end;
-
+	char* end = NULL;
 	*result = strtoull(str, &end, base);
-	if (*result == 0)
+	if (end == NULL)
```

Comment:
Why change this to check if end is NULL?

The documentation says that if no digits were found, `end` will be set to `str`, so this check for NULL should not work unless there is some platform specific quirk of which I am unaware:

https://www.man7.org/linux/man-pages/man3/strtoul.3.html

Reply:
Oh, good catch. Would you recommend checking `errno` here instead?

Feel free to squash / delete my commit with a more proper fix.

Reply:
I am not sure what the original issue was that prompted changing this, so it is hard to suggest what to do instead. That said, looking at some documentation, it is clear to me that this code probably is not correct:

https://docs.oracle.com/cd/E36784_01/html/E36886/ddi-strtoull-9f.html

The main issue that I see is that ddi_strtoull(“0”, NULL, 10, &res); will return whatever was in errno from a previous function.

To address that, you could do the same thing that they do in Illumos-gate:

https://github.com/illumos/illumos-gate/blob/9ecd05bdc59e4a1091c51ce68cce2028d5ba6fd1/usr/src/lib/libfakekernel/common/kmisc.c#L106

This looks like an old issue. I would suggest opening a PR against the main repository so we can try to avoid increasing the size of this diff.